pkg/resource/model_package/sdk.go

```diff
@@ -0,0 +1,1120 @@
+						}
+						if f11f0iter.TransformJobDefinition.TransformResources.InstanceType != nil {
+							f11f0elemf1f6.SetInstanceType(*f11f0iter.TransformJobDefinition.TransformResources.InstanceType)
+						}
+						if f11f0iter.TransformJobDefinition.TransformResources.VolumeKMSKeyID != nil {
+							f11f0elemf1f6.SetVolumeKmsKeyId(*f11f0iter.TransformJobDefinition.TransformResources.VolumeKMSKeyID)
+						}
+						f11f0elemf1.SetTransformResources(f11f0elemf1f6)
+					}
+					f11f0elem.SetTransformJobDefinition(f11f0elemf1)
+				}
+				f11f0 = append(f11f0, f11f0elem)
+			}
+			f11.SetValidationProfiles(f11f0)
+		}
+		if r.ko.Spec.ValidationSpecification.ValidationRole != nil {
+			f11.SetValidationRole(*r.ko.Spec.ValidationSpecification.ValidationRole)
+		}
+		res.SetValidationSpecification(f11)
+	}
+
+	return res, nil
+}
+
+// sdkUpdate patches the supplied resource in the backend AWS service API and
+// returns a new resource with updated fields.
+func (rm *resourceManager) sdkUpdate(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkUpdate")
+	defer exit(err)
+	if err = rm.requeueUntilCanModify(ctx, latest); err != nil {
+		return nil, err
+	}
+	input, err := rm.newUpdateRequestPayload(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *svcsdk.UpdateModelPackageOutput
+	_ = resp
+	resp, err = rm.sdkapi.UpdateModelPackageWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateModelPackage", err)
+	if err != nil {
+		return nil, err
+	}
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	if ko.Status.ACKResourceMetadata == nil {
+		ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	if resp.ModelPackageArn != nil {
+		arn := ackv1alpha1.AWSResourceName(*resp.ModelPackageArn)
+		ko.Status.ACKResourceMetadata.ARN = &arn
+	}
+
+	rm.setStatusDefaults(ko)
+	return &resource{ko}, nil
+}
+
+// newUpdateRequestPayload returns an SDK-specific struct for the HTTP request
+// payload of the Update API call for the resource
+func (rm *resourceManager) newUpdateRequestPayload(
+	ctx context.Context,
+	r *resource,
+) (*svcsdk.UpdateModelPackageInput, error) {
+	res := &svcsdk.UpdateModelPackageInput{}
+
+	if r.ko.Spec.ApprovalDescription != nil {
+		res.SetApprovalDescription(*r.ko.Spec.ApprovalDescription)
+	}
+	if r.ko.Spec.ModelApprovalStatus != nil {
+		res.SetModelApprovalStatus(*r.ko.Spec.ModelApprovalStatus)
+	}
+	if r.ko.Status.ACKResourceMetadata != nil && r.ko.Status.ACKResourceMetadata.ARN != nil {
+		res.SetModelPackageArn(string(*r.ko.Status.ACKResourceMetadata.ARN))
+	} else {
+		res.SetModelPackageArn(rm.ARNFromName(*r.ko.Spec.ModelPackageName))
```

Comment:
Update doesn't exist for unversioned Modelpackage. Not sure what logic needs to be added for versioned Model case since an ARN will always exists for them when going through sdkfind. And update call would come after the create which populates the ARN

Reply:
Looks like it is able to adopt both resources. Although when deleting the versioned modelpackage through kubectl it removes it from k8s side but still exists on sagemaker.

Reply:
ok for now to leave this as-is but please create a github issue and link this conversation, basically the else block should not be generated if ARN cannot be constructed from name or throw an an error like, missingRequiredField